src/MultiOwnerLightAccountFactory.sol

```diff
@@ -0,0 +1,113 @@
+    /// @return ret The address of either the newly deployed account or an existing account with these owners and salt.
+    function createAccount(address[] calldata owners, uint256 salt) public returns (MultiOwnerLightAccount ret) {
+        address addr = getAddress(owners, salt);
+        uint256 codeSize = addr.code.length;
+        if (codeSize > 0) {
+            return MultiOwnerLightAccount(payable(addr));
+        }
+        ret = MultiOwnerLightAccount(
+            payable(
+                new ERC1967Proxy{salt: bytes32(salt)}(
+                    address(accountImplementation), abi.encodeCall(MultiOwnerLightAccount.initialize, (owners))
+                )
+            )
+        );
+    }
+
+    /// @notice Create an account, and return its address. Returns the address even if the account is already deployed.
+    /// @dev This method uses less calldata than `createAccount` when creating accounts with a single initial owner.
+    /// @param owner The owner of the account to be created.
+    /// @param salt A salt, which can be changed to create multiple accounts with the same owner.
+    /// @return ret The address of either the newly deployed account or an existing account with this owner and salt.
+    function createAccountSingle(address owner, uint256 salt) public returns (MultiOwnerLightAccount ret) {
+        address[] memory owners = new address[](1);
+        owners[0] = owner;
+        address addr = getAddress(owners, salt);
+        uint256 codeSize = addr.code.length;
+        if (codeSize > 0) {
+            return MultiOwnerLightAccount(payable(addr));
+        }
+        ret = MultiOwnerLightAccount(
+            payable(
+                new ERC1967Proxy{salt: bytes32(salt)}(
+                    address(accountImplementation), abi.encodeCall(MultiOwnerLightAccount.initialize, (owners))
+                )
+            )
+        );
+    }
+
+    /// @notice Calculate the counterfactual address of this account as it would be returned by `createAccount`.
+    /// @param owners The owners of the account to be created.
+    /// @param salt A salt, which can be changed to create multiple accounts with the same owners.
+    /// @return The address of the account that would be created with `createAccount`.
+    function getAddress(address[] memory owners, uint256 salt) public view returns (address) {
+        _validateOwnersArray(owners);
+
+        return Create2.computeAddress(
+            bytes32(salt),
+            keccak256(
+                abi.encodePacked(
+                    type(ERC1967Proxy).creationCode,
+                    abi.encode(
+                        address(accountImplementation), abi.encodeCall(MultiOwnerLightAccount.initialize, (owners))
+                    )
+                )
+            )
+        );
+    }
+
+    /// @dev `owners` must be in strictly ascending order and not include the 0 address. Its length must not be empty
```

Comment:
Maybe add a comment about how the sorting ensures a canonical counterfactual for a given set of starting owners.